src/converter/bib/BibConversion.js

```diff
@@ -7,31 +7,122 @@ export function convertCSLJSON(source) {
   let result
 
-  if (bibType === 'article-journal') {
-    result = _convertJournalArticleFromCSLJSON(source)
+  // CSL types: http://docs.citationstyles.org/en/stable/specification.html#appendix-iii-types
+  let typeMapping = {
+    "article": "journal-article",
+    "article-magazine": "journal-article",
+    "article-newspaper": "journal-article",
+    "article-journal": "journal-article",
+    //"bill"
+    "book": "book",
+    //"broadcast"
+    "chapter": "book",
+    "dataset": "data-publication",
+    //"entry"
+    "entry-dictionary": "book",
+    "entry-encyclopedia": "book",
+    //"figure"
+    //"graphic"
+    //"interview"
+    //"legislation"
+    //"legal_case"
+    //"manuscript"
+    //"map"
+    //"motion_picture"
+    //"musical_score"
+    //"pamphlet"
+    "paper-conference": "conference-proceeding",
+    "patent": "patent",
+    //"post"
+    //"post-weblog"
+    //"personal_communication"
+    "report": "report",
+    //"review"
+    //"review-book"
+    //"song"
+    //"speech"
+    "thesis": "thesis",
+    //"treaty"
+    "webpage": "webpage"
+    //NA : "periodical"
+    //NA : "clinical-trial"
+    //NA : "preprint"
+    //NA : "software"
+  }
+
+  if (typeMapping[bibType]) {
+    result = _convertFromCSLJSON(source, typeMapping[bibType])
   } else {
     throw new Error(`Bib type ${bibType} not yet supported`)
   }
   return result
 }
 
-function _convertJournalArticleFromCSLJSON(source) {
+function _convertFromCSLJSON(source, type) {
   const date = _extractDateFromCSLJSON(source)
 
   let data = {
-    type: 'journal-article',
-    authors: source.author.map(a => {return {surname: a.family, givenNames: a.given}}),
-    articleTitle: source.title,
-    source: source['container-title'],
+    type: type,
+
     volume: source.volume,
     issue: source.issue,
     pageRange: source.page,
     doi: source.DOI,
+    pmid: source.PMID,
+
+    edition: source.edition,
+    publisherLoc: source['publisher-place'],
+    publisherName: source.publisher,
+    pageCount: source['number-of-pages'],
+    isbn: source.ISBN,
+
     year: date.year,
     month: date.month,
-    day: date.day
+    day: date.day,
+
+    uri: source.URL,
+    version: source.version
+
+    /* Examples with no corresponding field:
+        - abstract
+        - accessed
+        - collection-title
+        - composer
+        - director
+        - ISSN
+        - language
+        - number-of-volumes
+        - PMCID
+        - title-short
+        - translator
+    */
   }
 
+  // Mapping of title depends on item type
+  if (source.type==='book' || source.type==='report') {
+    data.source = source.title
+  } else if (source.type==='webpage') {
+    data.title = source.title
+  } else {
+    data.source = source['container-title']
+    if (source.type==='chapter') {
+      data.chapterTitle = source.title
+    } else if (source.type==='dataset') {
+      data.dataTitle = source.title
+    } else {
+      data.articleTitle = source.title
+    }
```

Comment:
Is there any reason for all these different labels for the title? I.e. `title`, `chapterTitle`, `dataTitle`, `articleTitle` for different item types. It would be much easier to have only one title and a possible container title...

Reply:
Agree we could normalise that in our internal model (EntityDatabse.js) and just call it `title` (needs adjustment of the importer/exporter code).

What do you mean by container title?

Reply:
Some publications are part of a container resource, e.g. a chapter is part of a book (i.e. the book title would be the container title) and a journal article is part of a journal (i.e. the journal title would be the container title). In German such publications are also named [unselbstständige Werke](https://de.wikipedia.org/wiki/Unselbst%C3%A4ndige_Literatur). I am not sure whether for the end user something like "container title" is clear and therefore maybe "book title" and "journal name" are better to present in the GUI.

Reply:
Understand. Yeah I guess in future containers and sub publications should even be linked together in a standardised way. However for now I guess we just accept the redundancy and treat them separately. We have a similar situation with affiliations as there often the university is the same but the department is different. However we don't model that hierarchy.

Reply:
No problem to have (possibly) some redundancy. And that is actually the same as other bibliographic data formats do like CSL-JSON, RIS, etc. do. Moreover, the possiblity in BibTeX to use links, e.g. [here](http://dblp.uni-trier.de/rec/bibtex/conf/post/NematiBGD18), is IMO rather uncommon practice. And as long as people can rely on some automatic way to receive all the data, they don't care about the details in the background;-)

The same affiliation might be much more likely to occur...